### PR TITLE
Add `NvidiaGenerator`

### PR DESCRIPTION
**Pull request by silvanocerza**

### CODE REVIEW

File: integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        model: Union[str, NvidiaGeneratorModel],
+        api_key: Secret = Secret.from_env_var("NVIDIA_API_KEY"),
+        model_arguments: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Create a NvidiaGenerator component.
+
+        :param model:
+            Name of the model to use for text generation.
+            See the [Nvidia catalog](https://catalog.ngc.nvidia.com/ai-foundation-models) to know the  supported models.
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py:61]:
```suggestion
            See the [Nvidia catalog](https://catalog.ngc.nvidia.com/ai-foundation-models) 
            for more information on the supported models.
```

File: integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            headers={
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+        )
+
+        if self._model not in SUPPORTED_MODELS:
+            models = ", ".join(e.value for e in NvidiaGeneratorModel)
+            msg = f"Model {self._model} is not supported, available models are: {models}"
+            raise ValueError(msg)
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py:91]: Why do we need this? The `from_str` call will automatically handle any unknown models.

File: integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    def run(self, prompt: str):
+        """
+        Queries the model with the provided prompt.
+
+        :param prompt:
+            Text to be sent to the generative model.
+        :returns:
+            A dictionary with the following keys:
+            - "replies": replies generated by the model.
+            - "meta": metadata for each reply.
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py:136]:
```suggestion
            - `replies` - Replies generated by the model.
            - `meta` - Metadata for each reply.
```

File: integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        for choice in data.choices:
+            replies.append(choice.message.content)
+            meta.append(
+                {
+                    "role": choice.message.role,
+                    "finish_reason": choice.finish_reason,
+                    # The usage field is not part of each choice, so we use reuse it each time
+                    "completion_tokens": data.usage.completion_tokens,
+                    "prompt_tokens": data.usage.prompt_tokens,
+                    "total_tokens": data.usage.total_tokens,
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py:158]: Let's just provide these as a separate `usage` key in the outputs (as their show the cumulative usage).

**Review by shadeMe — CHANGES_REQUESTED** [review]

File: integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py (hunk 2 above)
Comment (silvanocerza) [reply integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py:91]: It's a leftover from a previous implementation, I'll change it.

**Pull request merged**